### ISSUE TITLE
Add --output-file flag

### ISSUE THREAD

**Issue by mcky** [stylelint/stylelint#3586]
> What is the problem you're trying to solve?

We would like to save results of linting to store as artefacts in our CI pipeline. Eslint allows custom formatters and saving the output to a file, which is handy.

> What solution would you like to see?

A new CLI option (`-o`, `--output-file'`) which can be combined with a formatter to save the formatted results to file.

**Comment by jeddy3** [comment]
@mcky Thanks for the request and for using the template.

I can see how this flag could be handy. I'll label up as "help wanted", so feel free to contribute a PR.

**Comment by frytyler** [comment]
Any pointers where to start here? I took a peak through the `lib` directory but didn't make much progress. 

Is this functionality available at all and just needs to be added to the CLI or is it completely new?

**Comment by jeddy3** [comment]
> Is this functionality available at all and just needs to be added to the CLI or is it completely new?

I believe it's completely new. However, I think you can already achieve the same thing with using `>` on the command line (as [in the docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md#examples)), so this is just a feature for convenience:

```shell
stylelint "foo/*.css" > myTestReport.txt
```

The node API users can do something similar too.

**Comment by mishanga** [comment]
Output redirection is not useful in CI. Here is an example of using stylelint with lint-staged:
```javascript
const CI = process.env.CI;
module.exports = {
    linters: {
        '*.{js,jsx,ts,tsx}': [`eslint --no-ignore ${ CI ? '--quiet --output-file report-eslint.txt' : '' }`],
        '*.{css,scss}': [`${ CI ? 'sh -c "stylelint $* > $0" report-stylelint.txt' : 'stylelint'}`],
    },
};
```

**Comment by evilebottnawi** [comment]
PR welcome :+1:
